tmt/steps/prepare/feature.py

```diff
@@ -0,0 +1,197 @@
+        """ Get guest distro by parsing the guest facts """
+        distro = self.guest.facts.distro
+        if distro is None:
+            return None
+
+        if re.search(r'Fedora Linux \d+', distro):
+            return 'Fedora'
+        if re.search(r'CentOS Linux.*7\.', distro):
+            return 'CentOS-7'
+        if re.search(r'CentOS Stream.*8', distro):
+            return 'CentOS-Stream-8'
+        if re.search(r'CentOS Stream.*9', distro):
+            return 'CentOS-Stream-9'
+        if re.search(r'Red Hat Enterprise Linux.*8\.', distro):
+            return 'RHEL-8'
+        if re.search(r'Red Hat Enterprise Linux.*9\.', distro):
+            return 'RHEL-9'
+        return None
+
+
+class ToggleableFeature(Feature):
+    def enable(self, guest: Guest, logger: tmt.log.Logger) -> None:
+        raise NotImplementedError
+
+    def disable(self, guest: Guest, logger: tmt.log.Logger) -> None:
+        raise NotImplementedError
+
+
+class CRB(ToggleableFeature):
+    KEY = 'crb'
+    # TBD
+
+
+class FIPS(ToggleableFeature):
+    KEY = 'fips'
+    # TBD
+
+
+class EPEL(ToggleableFeature):
+    KEY = 'epel'
+
+    def enable(self, guest: Guest, logger: tmt.log.Logger) -> None:
+        guest_distro = self.get_guest_distro(guest=guest, logger=logger)
+        if guest_distro is None:
+            raise tmt.utils.PrepareError('The distro of the guest is not supported.')
+
+        if guest_distro == 'Fedora':
+            self.info('Enable EPEL on Fedora, do nothing ...')
+        elif guest_distro == 'CentOS-Stream-8':
+            # dnf config-manager --set-enabled powertools
+            # dnf install epel-release epel-next-release
+            self.info('Enable EPEL on CentOS Stream 8')
+            command = Command()
+            if self.guest.facts.is_superuser is False:
+                command += Command('sudo')
+            command1 = command + Command('dnf', 'config-manager', '--set-enabled', 'powertools')
+            command2 = command + Command('dnf', '-y', 'install',
+                                         'epel-release', 'epel-next-release')
+            self.guest.execute(command1, silent=True)
+            self.guest.execute(command2, silent=True)
+        elif guest_distro == 'RHEL-8':
+            # subscription-manager repos --enable codeready-builder-for-rhel-8-$(arch)-rpms
+            # dnf install https://dl.fedoraproject.org/pub/epel/epel-release-latest-8.noarch.rpm
+            self.info('Enable EPEL on RHEL 8')
+            command = Command()
+            if self.guest.facts.is_superuser is False:
+                command += Command('sudo')
+            command1 = command + \
+                Command('subscription-manager', 'repos', '--enable',
+                        'codeready-builder-for-rhel-8-$(arch)-rpms')
+            command2 = command + \
+                Command('dnf', '-y', 'install',
+                        'https://dl.fedoraproject.org/pub/epel/epel-release-latest-8.noarch.rpm')
```

Comment:
URL and rpm name pattern above, plus also repo names in fedora branch - I would move these magical constants to top level and define variables to hold them:

```python
FEDORA_REPO_NAME = 'powertools'
FEDORA_PACKAGE_NAMES = ['epel-release', 'epel-next-release']

RHEL_8_REPO_NAME = 'codeready-builder-for-rhel-8-$(arch)-rpms'
RHEL_8_PACKAGE_NAMES = ['https://dl.fedoraproject.org/pub/epel/epel-release-latest-8.noarch.rpm']


f'dnf -y install {" ".join(FEDORA_PACKAGES)}'
```

Again, more verbose, but these constants tend to change, it's usually better to make them more visible.